### DISCUSSION TITLE
Invalid Interception Error on Vercel but not Docker

### DISCUSSION THREAD

**Discussion by gwkline** [vercel/next.js#63946] [Help]
### Summary

Hi there - encountering some weirdness in production with route interception / parallel routes. Keep in mind this is not happening in a local Docker container environment, only when running on the Edge runtime on Vercel.

In one of my app router routes, say `/some-route-root/apples`, I have the following structure
```
.
├── @id
│   ├── (.)[id]
│   │   └── page.tsx
│   └── default.tsx
├── [id]
│   ├── notes
│   │   └── page.tsx
│   └── page.tsx
├── layout.tsx
└── page.tsx
```

My goal is that when the user navigates to `/some-route-root/apples/123`, the content under the intercepted `@id` directory is shown (in this case it's a sheet/drawer style component) on top of the current page.

Locally, this works. Navigating to `/some-route-root/apples/123` properly shows the drawer on top of the `/apples` page. 

In production, I see the following error in logs (hitting the function `/some-route-root/apples/(.)[id]`):
```
⨯ Error: Invalid interception route: /some-route-root/apples/(.). Must be in the format /<intercepting route>/(..|...|..)(..)/<intercepted route>
    at (node_modules/next/dist/esm/server/future/helpers/interception-routes.js?d73c:23:0)
    at (node_modules/next/dist/esm/shared/lib/router/utils/is-dynamic.js:6:52)
    at (node_modules/next/dist/esm/server/future/route-matcher-managers/default-route-matcher-manager.js:187:27)
    at (node_modules/next/dist/esm/server/base-server.js:1875:0)
    at (node_modules/next/dist/esm/server/base-server.js:1861:0)
    at (node_modules/next/dist/esm/server/lib/trace/tracer.js?07c4:79:0)
    at (node_modules/next/dist/esm/server/base-server.js:1855:24)
    at (node_modules/next/dist/esm/server/base-server.js:905:0)
    at (node_modules/next/dist/esm/server/base-server.js:842:0)
```

To be fair I'm not the most well versed in route interception/parallel routes, so it's very possible I'm doing something wrong. With that being said, I'd love to know why this is happening, and ideally how I can have the same behavior locally and in production. Please let me know if you have any ideas, thanks!

### Additional information

```js
Docker `next info`:

Operating System:
  Platform: linux
  Arch: arm64
  Version: #1 SMP Fri Feb 16 11:54:02 UTC 2024
  Available memory (MB): 8936
  Available CPU cores: 4
Binaries:
  Node: 20.12.0
  npm: 10.5.0
  Yarn: 1.22.19
  pnpm: N/A
Relevant Packages:
  next: 14.2.0-canary.51 // Latest available version is detected (14.2.0-canary.51).
  eslint-config-next: N/A
  react: 18.3.0-canary-2f8f77602-20240229
  react-dom: 18.2.0
  typescript: 5.3.3
Next.js Config:
  output: N/A
```

**Comment by zipme** [comment]
I have the same issue, local build and next start works fine, but got this Error: Invalid interception route error on vercel
